ping/ping_common.c

```diff
@@ -963,3 +965,18 @@ inline int is_ours(struct ping_rts *rts, socket_st * sock, uint16_t id)
 {
 	return sock->socktype == SOCK_DGRAM || id == rts->ident;
 }
+
+char *str_interval(int interval)
+{
+	static char buf[7];
+
+	/*
+	 * Here we always want to use locale regardless USE_IDN or ENABLE_NLS,
+	 * because we need do to print dot as decimal point of a number in -i option.
+	 */
+	setlocale(LC_ALL, "C");
+	sprintf(buf, "%.g", (double)interval / 1000);
+	setlocale(LC_ALL, "");
+
```

Comment:
If we wanted avoid messing with locales and floating point we can do something as:

```
        if (interval % 1000)
                printf("%1i.%03i", interval/1000, interval%1000);
        else
                printf("%i", interval/1000);
```

Reply:
I got the idea. We need `char *`, due `error()`, thus:

```C
static char buf[14];
if (interval % 1000)
         sprintf(buf, "%1i.%03i", interval/1000, interval%1000);
else
         sprintf(buf, "%i", interval/1000);

return buf;
```

Reply:
Maybe snprintf(buf, sizeof(buf), ...) would be safer, on the other hand as long as interval is 32bit int it cannot overflow 14 characters...

Reply:
Yep, it should be safe, but let's use `snprintf()` for extra security. Thanks!